src/core/thread/link_quality.hpp

```diff
@@ -211,6 +211,55 @@ class RssAverager
     uint16_t mCount : 5;    // Number of RSS values added to averager so far (limited to 2^kCoeffBitShift-1).
 };
 
+/**
+ * This class implements a LinkQuality Indicator (LQI) averager.
+ *
+ * It maintains the exponential moving average value of LQI.
+ *
+ */
+class LqiAverager
+{
+public:
+    /**
+     * This method reset the averager and clears the average value.
+     *
+     */
+    void Reset(void);
+
+    /**
+     * This method adds a link quality indicator (LQI) value to the average.
+     *
+     * @param[in] aLqi                Link Quality Indicator value to be added to the average.
+     *
+     */
+    void Add(uint8_t aLqi);
+
+    /**
+     * This method returns the current average link quality value maintained by the averager.
+     *
+     * @returns The current average value.
+     *
+     */
+    uint8_t GetAverage(void) const { return mAverage; }
+
+    /**
+     * This method returns the count of frames calculated so far.
+     *
+     * @returns The count of frames calculated.
+     *
+     */
+    uint8_t GetCount(void) const { return mCount; }
+
+private:
+    enum
+    {
+        kCoeffBitShift = 3, // Coefficient used for exponentially weighted filter (1 << kCoeffBitShift).
+    };
+
+    uint8_t mAverage; // The average link quality indicator value.
```

Comment:
same for other fields.
```suggestion
    uint8_t mAverage; ///< The average link quality indicator value.
```